### PR TITLE
dialects: (dsp) add dsp dialect

### PR DESCRIPTION
**Pull request by alecerio**
Add DSP dialect and STFT operation (with tests).

@zero9178 @math-fehr @compor

### CODE REVIEW

File: tests/filecheck/dialects/onnx/onnx_invalid.mlir [hunk 1]
```diff
@@ -551,3 +551,12 @@ builtin.module {
...
 }
+
+// -----
+
+builtin.module {
+  %t0 = "test.op"() : () -> (tensor<2x256xf32>)
+  %t1 = "arith.constant"() {"value" = 1 : i64}: () -> i64
+  // CHECK: Operation does not verify: frame number of dimensions must be 1. Actual number of dimensions: 2
+  %res_stft = "dsp.STFT"(%t0, %t1) {"frame_size" = 256 : i64, "hop_size" = 128 : i64}: (tensor<2x256xf32>, i64) -> tensor<2x128xf32>
+}
```
Comment (tobiasgrosser) [inline tests/filecheck/dialects/onnx/onnx_invalid.mlir:562]: Why is this added into the onnx folder and not into a new dsp folder?

Comment (alecerio) [reply tests/filecheck/dialects/onnx/onnx_invalid.mlir:562]: Yes, definitely the wrong place

**Comment by superlopuh** [comment]
Hi @alecerio, I'll close this PR for now, let's reopen it if you decide to pick up the project again
